src/Language/PureScript/Renamer.hs

```diff
@@ -175,7 +177,7 @@ renameInLiteral _ l = return l
 -- Renames within case alternatives.
 --
 renameInCaseAlternative :: CaseAlternative Ann -> Rename (CaseAlternative Ann)
-renameInCaseAlternative (CaseAlternative bs v) =
+renameInCaseAlternative (CaseAlternative bs v) = newScope $
```

Comment:
Yep, that makes total sense. I assume this is where the spurious overlap warnings were coming from?